src/test/java/com/csoft/MainMojoTest.java

```diff
@@ -0,0 +1,146 @@
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class MainMojoTest {
+    private final String goodLicense = "Happy Freedom License v1";
+    private final String badLicense = "Bad Banned License v2";
+
+    private final Set<Artifact> empty = Collections.emptySet();
+    @Rule
+    public TestResources testResources = new TestResources();
+    @Rule
+    public final TestMavenRuntime maven = new TestMavenRuntime();
+    private TestLog log;
+    private TestProjectBuilder builder;
+
+    @Before
+    public void setUp() throws Exception {
+        builder = new TestProjectBuilder();
+    }
+
+    @Test
+    public void basicTestWithNoDependencies() throws Exception {
+        MainMojo mojo = configure(empty, empty);
+
+        mojo.execute();
+
+        log.assertInfo(" - artifactId          : test-project");
+        log.assertInfo(" - groupId             : com.acme.test.co");
+        log.assertInfo(" - description         : A nice test pom");
+        log.assertInfo(" - version             : 1");
+        log.assertInfo(" - getArtifact.activeP : []");
+        log.assertInfo(" - getArtifact.artId   : test-project");
+        log.assertInfo(" - getArtifact.groupId : com.acme.test.co");
+        log.assertInfo(" - getArtifact.version : 1");
+        log.assertInfo(" - getArtifacts.isEmpty: true");
+    }
+
+    @Test
+    public void passesWithAllGoodLicenses() throws Exception {
+        MainMojo mojo = configure(
+                builder.createArtifact("acme",
+                    "main",
+                    "2",
+                    goodLicense),
+                builder.createArtifact("acme",
+                        "artifact",
+                        "1",
+                        goodLicense));
+
+        mojo.execute();
+
+        log.assertNoWarning("Found 1 violations for license 'Bad Banned License v2':");
+    }
+
+    @Test
+    public void willFailWhenBannedLicenseShowsUpInTransient() throws Exception {
+        MainMojo mojo = configure(empty,
+                builder.createArtifact("acme",
+                        "artifact",
+                        "1",
+                        badLicense));
+
+        try {
+            mojo.execute();
+            fail("should have thrown error");
+        } catch (MojoFailureException e) {
+            assertEquals("Failing build", e.getMessage());
+            log.assertWarning("Found 1 violations for license 'Bad Banned License v2':");
+            log.assertWarning(" - acme:artifact:1:null");
+        }
+    }
+
+    /*
+     * The following are a set of characterization tests that I think are probably bugs
+     * These tests clarify current behavior for discussion
+     * If these are desired, simply remove this comment.
+     * */
+    @Test
+    public void doesNotFailUnlessPrintLicensesIsOn() throws Exception {
```

Comment:
love this. yes it's definitely a bug, let's fix it in another PR, and it will be super nice to keep the test anyway(and change the expected result =)